utils/stall_free.hh

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <list>
+#include <algorithm>
+#include <seastar/core/thread.hh>
+
+namespace utils {
+
+// Similar to std::merge but it does not stall. Must run inside a seastar
+// thread. Merge items from list1 and list2 into list_output. Items from list1
+// can be moved. Items from list2 can only be copied.
+template<class T, class Compare>
+std::list<T> stall_free_merge(std::list<T>& list1, const std::list<T>& list2, Compare comp) {
```

Comment:
list1 should be an rvalue since it is destroyed in the process.

alternatively, call it merge_to(list<T>& to, const list<T>& from); and then you don't have to move from list1 to the result.

Reply:
I changed to merge_to.